gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -681,18 +681,20 @@ private void cancelDag(DagId dagId) throws ExecutionException, InterruptedExcept
       clearUpDagAction(dagId);
     }
 
-    private void cancelDagNode(DagNode<JobExecutionPlan> dagNodeToCancel) throws ExecutionException, InterruptedException {
+    private void cancelDagNode(DagNode<JobExecutionPlan> dagNodeToCancel, boolean shouldSendCancellationEvent) throws ExecutionException, InterruptedException {
```

Comment:
can you add a comment to explain why you added this boolean and what case we don't send cancellation event even though this function is called?